crypto/build.rs

```diff
@@ -0,0 +1,17 @@
+extern crate gcc;
+
+use std::env;
+use std::path::Path;
+
+fn main() {
+    gcc::Config::new()
+        .compiler(&Path::new("arm-none-eabi-gcc"))
+        .flag("-mcpu=cortex-m4")
```

Comment:
So this is odd. The assembly you have is for `cortex-m0`, but you're compiling for `cortex-m4`.